0035-ilp-over-http/0035-ilp-over-http.md

```diff
@@ -24,26 +25,34 @@ This is a minimal protocol built on HTTP. HTTP/2 is HIGHLY RECOMMENDED for perfo
 
 Peers MAY use any standard HTTP authentication mechanism to authenticate incoming requests. TLS Client Certificates are RECOMMENDED between peers for security and performance, though bearer tokens such as JSON Web Tokens (JWTs) or Macaroons MAY be used instead. Basic authentication (username and password) is NOT RECOMMENDED, because of the additional delay introduced by securely hashing the password.
 
-### Request
+### ILP Prepare
+
+#### Request
 
 ```http
 POST /ilp HTTP/x.x
 Host: example.com
 Accept: application/octet-stream
 Content-Type: application/octet-stream
 Authorization: Bearer zxcljvoizuu09wqqpowipoalksdflksjdgxclvkjl0s909asdf
+Idempotency-Key: 8988dd17-55e4-40e0-9c57-419d81a0e3a5
 
 < Body: Binary OER-Encoded ILP Prepare Packet >
 ```
 
-**Field Details:**
+- **Path** &mdash; A connector MAY specify any HTTP path for their peer to send ILP packets to.
+- **Host Header** &mdash; The standard [HTTP Host Header](https://tools.ietf.org/html/rfc2616#section-14.23) indicating the domain of the HTTP server the Request is sent to.
+- **Content-Type / Accept Headers** &mdash; MUST be set to `application/octet-stream`.
+- **Idempotency Key Header** &mdash; Globally unique string identifying this request, which MUST be derived from a cryptographically secure source of randomness.
+- **Body** &mdash; ILP Packet encoded using OER, as specified in [RFC 27: Interledger Protocol V4](./0027-interledger-protocol-4/0027-interledger-protocol-4.md).
 
-- **Path** - A connector MAY specify any HTTP path for their peer to send ILP packets to.
-- **Host Header** - The standard [HTTP Host Header](https://tools.ietf.org/html/rfc2616#section-14.23) indicating the domain of the HTTP server the Request is sent to.
-- **Content-Type / Accept Headers** - MUST be set to `application/octet-stream`.
-- **Body** - ILP Packet encoded using OER, as specified in [RFC 27: Interledger Protocol V4](./0027-interledger-protocol-4/0027-interledger-protocol-4.md).
+#### Response
+
+```http
+HTTP/x.x 200 OK
+```
 
-### Response
+For backwards compatibility with [RFC 35: ILP over HTTP (draft 2)](https://interledger.org/rfcs/0035-ilp-over-http/draft-2.html), if no `Idempotency-Key` header is included, the raw OER-encoded ILP Fulfill or Reject MUST be returned within the body of the response:
```

Comment:
Would the previous method be supported forever or would the intention be to deprecate it in the future?

Reply:
While I don't think anyone would want to use the old version in production, there are a couple advantages to it:
1. Much easier for manual debugging
2. Doesn't require a publicly accessible endpoint to send outgoing packets

I'm indifferent, since (1) could be addressed by adding an admin endpoint to the node that makes it easier to send packets for debugging, and (2) might be best addressed using BTP or a WebSocket-based transport.

Should the old method eventually be deprecated?

Reply:
2) is something that we may consider in Coil's use case, where you have clients who only want to send packets. We use BTP right now which is OK but switching to HTTP could save on some connection logic. That said there's really no disadvantage to using nonstandard ILP HTTP when we have to so it's OK if this doesn't cover every use case for everyone

Reply:
I echo Ben's sentiment - I think we want to allow for ILP-over-HTTP to be used by a non-server in a "send-only" mode (e.g., a mobile app). I added a proposed change.